presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -472,14 +473,14 @@ public boolean isDone()
     public boolean transitionToPlanning()
     {
         queuedTime.compareAndSet(null, nanosSince(createNanos).convertToMostSuccinctTimeUnit());
+        totalPlanningStartNanos.compareAndSet(null, System.nanoTime());
         return queryState.compareAndSet(QUEUED, PLANNING);
     }
 
     public boolean transitionToStarting()
     {
-        Duration durationSinceCreation = nanosSince(createNanos).convertToMostSuccinctTimeUnit();
-        queuedTime.compareAndSet(null, durationSinceCreation);
-        totalPlanningTime.compareAndSet(null, durationSinceCreation);
+        queuedTime.compareAndSet(null, nanosSince(createNanos).convertToMostSuccinctTimeUnit());
+        recordPlanningTime();
```

Comment:
This code was duplicated and it took me failed test to realize that there are several occurrences. For sakes for maintainability, let us have it in one place. Focused named method never a bad idea - especially, it self documents with name record planning time). Probably offline.